### ISSUE TITLE
GLViewWidget::readQImage() partial image on hidpi

### ISSUE THREAD

**Issue by pijyoi** [pyqtgraph/pyqtgraph#1732]
### Short description

On a system with device pixel ratio of 2.0, GLViewWidget::readQImage() does not save the full screen image. Instead, about a quarter is saved.

### Code to reproduce

```python
import pyqtgraph.opengl as gl
import pyqtgraph as pg
import numpy as np

app = pg.mkQApp()
w = gl.GLViewWidget()
w.opts['distance'] = 30
w.show()

n = 101
s = np.linspace(-10,10,n)
X, Y = np.meshgrid(s, s)
Z = 10 * np.sinc(np.hypot(X, Y))
for i in range(100):
    color = pg.glColor((i,n*1.3))
    row = np.column_stack((X[i,:], Y[i,:], Z[i,:]))
    col = np.column_stack((X[:,i], Y[:,i], Z[:,i]))
    plt = gl.GLLinePlotItem(pos=row, color=color)
    w.addItem(plt)
    plt = gl.GLLinePlotItem(pos=col, color=color)
    w.addItem(plt)

app.exec_()
qimg = w.readQImage()
qimg.save('qimage.png')

```

### Expected behavior

![expected](https://user-images.githubusercontent.com/2657027/115569214-9673ff80-a2ef-11eb-9524-fd304ccba528.png)

### Real behavior

![qimage](https://user-images.githubusercontent.com/2657027/115568985-63317080-a2ef-11eb-9772-4c4b173d2c8c.png)

### Tested environment(s)

 * PyQtGraph version: 0.12.1 
 * Qt Python binding: PySide and PyQt 
 * Python version: 3.8.9
 * NumPy version: 1.20.2 
 * Operating system: Windows 10
 * Installation method: pip install -e . 

### Additional context
Verified that on a non-hidpi system, the full screen image is saved.

I don't quite understand the following code found in GLViewWidget.py
```python
    def width(self):
        if self._dpiRatio.is_integer():
            return super().width()
        else:
            return int(super().width() * self._dpiRatio)
```
If dpr is 2.0, it takes the first branch.
If dpr is 1.5, it would take the second branch.
i.e. if I were to multiply width() by dpr to get the full image width, wouldn't that cause double multiplication when dpr is 1.5?

**Comment by j9ac9k** [comment]
I ran into this when testing my hidpi settings and noticed when scaling was set to a non fractional value.  I had written some code to compensate for that; but I think that code got removed in a later PR.  I'll investigate later today.

**Comment by j9ac9k** [comment]
> I don't quite understand the following code found in GLViewWidget.py

Forgot to follow up now that I'm at a keyboard.

During my testing for #1516 I noticed that when scaling was at an even integer; the views would be as you see them in the screen shot; but at fractional scaling things was as expected.  I saw reference to this kind of logic in the Matplotlib repository, but on a quick google search, I couldn't find it.

I wonder if this issue came up again with the migration to QOpenGLWidget in #1525

**Comment by pijyoi** [comment]
In my limited testing, #1735 seems to fix this issue.

Using the environment variable QT_SCALE_FACTOR, I seem to be able to simulate fractional dpi ratios.
e.g. original dpr being 2.0, with QT_SCALE_FACTOR=1.25, reported dpr will be 2.5 to the application.

For both integer (2.0) and fractional (2.5), the full image gets saved with image sizes (1280x960) and (1600x1200) respectively.

**Comment by j9ac9k** [comment]
Which platform did you test on?  I can test on macOS and windows; I have hidpi and non-hidpi monitors at my disposal.

**Comment by pijyoi** [comment]
>Which platform did you test on?

So far, just Windows 10 which has a dpr 2.0 display, Qt 5.15 and Qt 6.0.
I could test against PyQt5 5.12 (but not PySide2 5.12) since there are wheels available for Python 3.8.

My Linux system only has a dpr 1.0 display. Although using QT_SCALE_FACTOR for simulation could be sufficient.

**Comment by j9ac9k** [comment]
On Windows 10, on my HiDPI display with PySide2 5.12.6 (on python 3.7) everything works as intended; I think the PR you submitted is good to merge; but I don't mind testing on more combinations/platforms if think we should.

Unrelated to the issue you're trying to solve, but I got this error on pyside2-512 + pyopengl 3.1.5 on python 3.7

```
Traceback (most recent call last):
  File ".\glqimage.py", line 24, in <module>
    qimg = w.readQImage()
  File "C:\Users\ogi\source\repos\j9ac9k\pyqtgraph\pyqtgraph\opengl\GLViewWidget.py", line 558, in readQImage
    glReadPixels(0, 0, w, h, GL_RGBA, GL_UNSIGNED_BYTE, pixels)
  File "C:\Users\ogi\source\repos\j9ac9k\pyqtgraph\.venv-pyside2_512-py37\lib\site-packages\OpenGL\GL\images.py", line 372, in glReadPixels
    imageData
  File "C:\Users\ogi\source\repos\j9ac9k\pyqtgraph\.venv-pyside2_512-py37\lib\site-packages\OpenGL\platform\baseplatform.py", line 415, in __call__
    return self( *args, **named )
  File "C:\Users\ogi\source\repos\j9ac9k\pyqtgraph\.venv-pyside2_512-py37\lib\site-packages\OpenGL\error.py", line 234, in glCheckError
    baseOperation = baseOperation,
OpenGL.error.GLError: GLError(
        err = 1282,
        description = b'invalid operation',
        baseOperation = glReadPixels,
        cArguments = (
                0,
                0,
                1280,
                960,
                GL_RGBA,
                GL_UNSIGNED_BYTE,
                c_void_p(1378184273984),
        )
)
```
